### ISSUE TITLE
Lease id validation for azure datalake gen 2

### ISSUE THREAD

**Issue by kahing** [Azure/azure-sdk-for-go#14798]
https://github.com/Azure/azure-sdk-for-go/blob/4f37ebbcd228a0b5c49eb2cced17c8fa2c737f55/services/storage/datalake/2018-11-09/storagedatalake/path.go#L353

How should I generate this lease id to pass the validation? The hadoop implementation uses `java.util.UUID`: https://github.com/apache/hadoop/blob/51991c4907262716e522e327edf96bea1fd3c494/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java#L400 . `java.util.UUID` is based on RFC 4122, but using https://pkg.go.dev/github.com/google/uuid#UUID.String which is implementing the same RFC results in an UUID that doesn't pass the validation.

I also cannot find similar validation in the hadoop's datalake gen2 implementation.

**Comment by ArcturusZhang** [comment]
Hi @kahing thanks for this issue!

The SDK was generated from the corresponding swagger, hence the validation logic. Therefore to remove that validation we will have the changes on the swagger first.
For workaround, you can use `validation.Disable = true` to disable the validation: https://github.com/Azure/go-autorest/blob/master/autorest/validation/validation.go#L28

**Comment by kahing** [comment]
@ArcturusZhang thanks will try that. It's a bit hard to parse the validation regex, do you also have an example of what would pass the validation?

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang thanks will try that. It's a bit hard to parse the validation regex, do you also have an example of what would pass the validation?

This regex stands for a UUID, with or without the hyphens, for instance, `00000000-0000-0000-0000-000000000000` fits this regex, and if you remove some or all of the hyphens in this one, it also fits this regex. In addition this regex allows you to add parenthesis around the value, for instance `(00000000-0000-0000-0000-000000000000)`, `{00000000-0000-0000-0000-000000000000}` can also fit the regex. It does not require your parenthesis to match, therefore `(00000000-0000-0000-0000-000000000000` or `{00000000-0000-0000-0000-000000000000)` should also pass the validation.
I am only parsing this regex by my eyes, therefore it could be not accurate. You could use this website to do some experiments: https://regex101.com/
